### PR TITLE
feat: Refactor additional salary

### PR DESCRIPTION
**Pull request by Anurag810**
> Please provide enough information so that others can review your pull request:

> Explain the **details** for making this change. What existing problem does the pull request solve?

> Screenshots/GIFs

### CODE REVIEW

File: erpnext/hr/doctype/additional_salary/additional_salary.py [hunk 1]
```diff
@@ -9,6 +9,13 @@
 from frappe.utils import getdate, date_diff
 
 class AdditionalSalary(Document):
+
+	def on_submit(self):
+		if self.ref_doctype == "Employee Advance" and self.ref_docname:
+			emp_adv = frappe.get_doc(self.ref_doctype, self.ref_docname)
+			emp_adv.return_amount = self.amount
+			emp_adv.save()
```
Comment (nabinhait) [inline erpnext/hr/doctype/additional_salary/additional_salary.py]: why not frappe.db.set_value?

File: erpnext/hr/doctype/employee_advance/employee_advance.js [hunk 2]
```diff
@@ -47,19 +55,38 @@ frappe.ui.form.on('Employee Advance', {
...
 	},
 
+	make_deduction_via_additional_salary: function(frm){
+		frappe.call({
+			method: "erpnext.hr.doctype.employee_advance.employee_advance.create_return_through_additional_salary",
+			args: {
+				doc: frm.doc
+			},
+			callback: function (r){
+				console.log("Helloxs")
```
Comment (nabinhait) [inline erpnext/hr/doctype/employee_advance/employee_advance.js]: remove console.log

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Review by nabinhait — CHANGES_REQUESTED** [review]
Patch is missing to migrate additional salary reference leave encashment, employee incentive and salary slip.

**Review by nabinhait — CHANGES_REQUESTED** [review]
Travis is failing

File: erpnext/hr/doctype/additional_salary/additional_salary.py (hunk 1 above)
Comment (Anurag810) [reply erpnext/hr/doctype/additional_salary/additional_salary.py]: done

File: erpnext/hr/doctype/employee_advance/employee_advance.js (hunk 2 above)
Comment (Anurag810) [reply erpnext/hr/doctype/employee_advance/employee_advance.js]: done

**Review by nabinhait — COMMENTED** [review]
Merged manually after fixing tests and patch

**Pull request merged**